### PR TITLE
fix(combobox): only open when text exists and filtered items are present

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #9617

## Summary

- if no filtered items are displayed, the combobox should not be opened
- adds test

### CODE REVIEW

File: packages/calcite-components/src/components/combobox/combobox.tsx [hunk 1]
```diff
@@ -1259,6 +1263,7 @@ export class Combobox
       }
       this.updateItems();
       this.filterItems("");
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.tsx:1265]: Can you use the new flag to open after filtering for the new value? Similar to [above](https://github.com/Esri/calcite-design-system/pull/9618/files#diff-9bc2dae4daf67ad95f668837f3083752245fbc22e0a58b1e5a043830e05af02cL1056-L1057)?

Comment (driskull) [reply packages/calcite-components/src/components/combobox/combobox.tsx:1265]: I don't think so. That one will only open when there is text present in the input and this one should open regardless.

Comment (jcfranco) [reply packages/calcite-components/src/components/combobox/combobox.tsx:1265]: Gotcha. Maybe the function argument can be renamed to distinguish this.

**Review by jcfranco — APPROVED** [review]
📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝
📝🔍📝📝📝🔍🔍🔍🔍📝🔍🔍🔍🔍🔍📝🔍📝📝📝🔍📝🔍📝
📝🔍📝📝📝🔍📝📝📝📝📝📝🔍📝📝📝🔍🔍📝🔍🔍📝🔍📝
📝🔍📝📝📝🔍📝🔍🔍📝📝📝🔍📝📝📝🔍📝🔍📝🔍📝🔍📝
📝🔍📝📝📝🔍📝📝🔍📝📝📝🔍📝📝📝🔍📝📝📝🔍📝📝📝
📝🔍🔍🔍📝🔍🔍🔍🔍📝📝📝🔍📝📝📝🔍📝📝📝🔍📝🔍📝
📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝

File: packages/calcite-components/src/components/combobox/combobox.tsx (hunk 1 above)
Comment (driskull) [reply packages/calcite-components/src/components/combobox/combobox.tsx:1265]: Yup, will do. 👍

**Pull request merged**